### PR TITLE
Logs: Do not insert log-line into log-fields in json download

### PR DESCRIPTION
**Pull request by gabor**
currently, when you download the logs as JSON, the original log-line will be part of the "fields" section for every log-line (so every log-line line will be there twice).

NOTE: in the tests in `public/app/features/logs/utils.test.ts` i made sure the linked dataframe is more complete, because the tested function will look into that frame to decide what to do.

how to test:
1. open some logs (for example in Loki or Elastic), click the [Download] button, choose `json`. check the downloaded JSON, verify that the log-line does not appear in the `fields` part
2. make sure correlations on the log-line still work:
    - enable the `correlations` feature-flag
    - create a correlation in the `administration/correlations` section. the important part is that in step3, you select the correct datasource, and enter the correct field-name. if you do the test with Loki, the field-name should be `Line`. 
    - now open some logs in explore, expand the log-details on a log-line, and verify that you can see the link created by the correlation

### CODE REVIEW

File: public/app/features/logs/components/logParser.test.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ describe('logParser', () => {
         dataFrame: new MutableDataFrame({
           refId: 'A',
           fields: [
+            testLineField,
```
Comment (gabor) [inline public/app/features/logs/components/logParser.test.ts:15]: now (correctly), we ignore the first string-field in the dataframe (which is the log-line-field). to make the current tests work, i add an explicit log-line-field, which will get removed.

File: public/app/features/logs/components/logParser.ts [hunk 2]
```diff
@@ -86,9 +86,26 @@ export const getDataframeFields = memoizeOne(
 );
 
 function shouldRemoveField(field: Field, index: number, row: LogRowModel) {
```
Comment (gabor) [inline public/app/features/logs/components/logParser.ts:88]: most of the changes in this file are just by re-ordering the tests.

File: public/app/features/logs/components/logParser.ts [hunk 3]
```diff
@@ -103,13 +120,12 @@ function shouldRemoveField(field: Field, index: number, row: LogRowModel) {
...
-  // hidden field
-  if (field.config.custom?.hidden) {
-    return true;
-  }
-  // field that has empty value (we want to keep 0 or empty string)
-  if (field.values[row.rowIndex] == null) {
+
+  // first string-field is the log-line
+  const firstStringFieldIndex = row.dataFrame.fields.findIndex((f) => f.type === FieldType.string);
+  if (firstStringFieldIndex === index) {
```
Comment (gabor) [inline public/app/features/logs/components/logParser.ts:126]: this is the "real" new code in the PR, the check for the log-line-field (first string-field in the dataframe)

**Review by svennergr — APPROVED** [review]
LGTM

**Pull request merged**